CRISPResso2/CRISPRessoShared.py

```diff
@@ -1661,3 +1672,267 @@ def zip_results(results_folder):
         )
     sb.call(cmd_to_zip, shell=True)
     return
+
+def safety_check(crispresso2_info, aln_stats, min_total_reads=10000, alignedCutoff=.9, alternateAlignment=.3, minRatioOfModsInToOut=.01, modificationsAtEnds=.01, outsideWindowMaxSubRate=.002, maxRateOfSubs=0.3, guide_len=19, amplicon_len=50, ampliconToReadLen = 1.5):
+    # If amplicon is significantly (1.5x) longer than reads
+    """Check the results of analysis for potential issues and warns the user.
+    Parameters
+    ----------
+    crispresso2_info : dict
+        Dictionary of values describing the analysis
+    aln_stats : dict
+        Dictionary of alignment statistics and modification frequency and location
+    min_total_reads : int
+        Cutoff value for total reads aligned
+    alignedCutoff : float
+        Check value for the percentage of reads aligned vs not aligned
+    alternateAlignment : float
+        Percentage of variance from expected reads to trigger guardrail
+    minRatioOfModsInToOut : float
+        Float representing the acceptable ratio of modifications in the window to out
+    modificationsAtEnds : float
+        The ratio of reads with modifications at the 0 or -1 spot
+    outsideWindowMaxSubRate : float
+        Ratio of subs allowed outside of the window
+    maxRateOfSubs : float
+        Allowed rate of subs accross the entire read
+    guide_len : int
+        Minimum guide length
+    amplicon_len : int 
+        Minimum guide length
+    ampliconToReadLen : float
+        Comparison value between amplicons and reads
+
+
+    Returns
+    -------
+    messages : list
+        List of messages returned from the violated guardrails to be included in the report.
+
+    """
+    logger = logging.getLogger(__name__)
+    logger.setLevel(logging.INFO)
+    messages = []
+
+    messageHandler = GuardRailMessageHandler(logger)
+    totalReadsGuardRail = TotalReadsGuardRail(messageHandler, min_total_reads)
+    total_reads = totalReadsGuardRail.safety(aln_stats['N_TOT_READS'])
+    if total_reads is not None:
+        messages.append(total_reads)
```

Comment:
What if we moved each of these checks into a new method for each `GuardRail` class? We can pass in `messages` and keep the logic of when to add the message together. Or we could pass `messages` to `.safety` and just handle it in there.

Reply:
This could also be the responsibility of the `report_warning` method in `GuardRailMessageHandler`.